source/grid/tria.cc

```diff
@@ -3099,28 +3106,25 @@ namespace internal
                               ExcLineInexistant(line_vertices.first,
                                                 line_vertices.second));
               }
-            // Assert that only exterior
-            // lines are given a boundary
-            // indicator
-            AssertThrow(
-              line->at_boundary(),
-              ExcInteriorLineCantBeBoundary(line->vertex_index(0),
-                                            line->vertex_index(1),
-                                            boundary_line->boundary_id));
-
-            // and make sure that we don't
-            // attempt to reset the
-            // boundary indicator to a
-            // different than the
-            // previously set value
-            if (line->boundary_id() != 0)
-              AssertThrow(line->boundary_id() == boundary_line->boundary_id,
-                          ExcMessage(
-                            "Duplicate boundary lines are only allowed "
-                            "if they carry the same boundary indicator."));
+            // Only exterior lines can be given a boundary indicator
+            if (line->at_boundary())
+              {
+                // make sure that we don't attempt to reset the boundary
+                // indicator to a different than the previously set value
+                if (line->boundary_id() != 0)
+                  AssertThrow(line->boundary_id() == boundary_line->boundary_id,
+                              ExcMessage(
+                                "Duplicate boundary lines are only allowed "
+                                "if they carry the same boundary indicator."));
 
-            line->set_boundary_id_internal(boundary_line->boundary_id);
+                line->set_boundary_id_internal(boundary_line->boundary_id);
+              }
             // Set manifold id if given
+            if (line->manifold_id() != numbers::flat_manifold_id)
+              AssertThrow(line->manifold_id() == boundary_line->manifold_id,
+                          ExcMessage(
+                            "Duplicate boundary lines are only allowed "
+                            "if they carry the same manifold indicator."));
```

Comment:
Is it clear at this point that the line is really at the boundary? It seems to me like you're already outside the `if (line->at_boundary())` block.

Reply:
No, the line is not necessarily on the boundary. Otherwise, I can't copy all the manifold ids.

Reply:
But then the error message and the comment above is wrong :-)